plugins/net_plugin/net_plugin.cpp

```diff
@@ -577,7 +577,7 @@ namespace eosio {
       std::atomic<bool>                     connecting{true};
       std::atomic<bool>                     syncing{false};
       uint16_t                              protocol_version = 0;
-      uint16_t                              consecutive_rejected_blocks = 0;
+      std::atomic<uint16_t>                 consecutive_rejected_blocks = 0;
```

Comment:
This does not need to be atomic, it is only accessed via connection strand.

Reply:
`sync_manager` also accesses it, though it uses a lock (that could possibly be moved to an inner scope if it's atomic).

Reply:
The key thing is it called from the connection strand. So it is only accessed via the connection strand so no lock or atomic needed.